src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -990,7 +1014,15 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                     LoopTerminatedDueToNonAsciiDataInVectorLocal:
 
                         outputBytesRemaining -= 8 * i;
-                        possibleNonAsciiQWord = Sse2.X64.ConvertToUInt64(utf16Data.AsUInt64());
+
+                        if (Sse2.X64.IsSupported)
+                        {
+                            possibleNonAsciiQWord = Sse2.X64.ConvertToUInt64(utf16Data.AsUInt64());
+                        }
+                        else
+                        {
+                            possibleNonAsciiQWord = utf16Data.AsUInt64().ToScalar();
```

Comment:
Question for @tannergooding  - Any reason why we can't just use `utf16Data.AsUInt64().ToScalar();` for both SSE2 and AdvSimd?

Reply:
`ToScalar()` wasn't optimized for integer types on x86 until https://github.com/dotnet/runtime/pull/37882.  It would be fine to use it now.

Reply:
Right, it should be fine to use now. We just hadn't optimized it yet for .NET Core 3

Reply:
Ok. So @carlossanlop, may be in follow-up PR you can try having ` possibleNonAsciiQWord = utf16Data.AsUInt64().ToScalar();` without an if-else.